File: Veri Yapilari/Stack/Stacks.py
class node:
    def __init__(self, incomingData=None):
        if (incomingData is not None):
            self.data = incomingData;
            self.prev = None;
        else:
            self.data = None;
            self.prev = None;
class Stack:
    def __init__(self, incomingSize):
        self.size = incomingSize
        self.root = node()
        self.SP = self.root
        self.capacity = self.size
    def stackPrinter(self, temp):
        if(temp.prev is None):
            return
        print(temp.data)
        temp = temp.prev
        self.stackPrinter(temp)
    def isFull(self):
        if(self.capacity == 0):
            return True
        else:
            return False
    def isEmpty(self):
        if (self.capacity == self.size):
            return True
        else:
            return False

    def  push(self,incomingData):
        if(self.isFull()):
            print("StackOverFlow!")
        else:
            temp = node(incomingData)
            temp.prev = self.SP
            self.SP = temp
            self.capacity = self.capacity - 1
    def display(self):
        self.stackPrinter(self.SP)
    def pop(self):
        if(self.isEmpty()):
            print("StackUnderFlow!")
        else:
            temp = self.SP
            self.SP = self.SP.prev
            self.capacity = self.capacity + 1
            return  temp.data

test = Stack(5)

for i in range(6):
    test.push(i+1)

test.display()

for i in range(6):
    test.pop()

File: Veri Yapilari/Stack/Stacks.cpp
#include "pch.h"
#include <iostream>
#include <time.h>
class MyObject {
public:
	int data;
	MyObject() = default;
	MyObject(int incomingData) : data(incomingData)
	{
	}

	friend std::ostream& operator<<(std::ostream& out, const MyObject& rhs);


	bool operator==(const MyObject& rhs)
	{
		if (this->data == rhs.data)
			return true;
		return false;
	}

};
std::ostream& operator<<(std::ostream& out, const MyObject& rhs)
{
	out << rhs.data;
	return out;
}

template <typename T>
class Stack
{
	struct node
	{
		T data;
		node* prev = {};
		node(T incomingData) :data(incomingData) {}
		node() = default;
		node(node&&) = default;
		node(const node&) = default;
		~node() = default;
	};
	node* root;
	node* SP;
	int size;
	int capacity;
public:
	Stack(int incomingSize)
	{
		this->root = nullptr;
		this->size = incomingSize;
		this->SP = this->root;
		this->capacity = this->size;
	}
	~Stack() {
		delete root;
	}
	void push(T incomingData) {
		if (this->isFull()) {
			std::cout << "StackOverFlow!\n";
		}
		else {
			node* temp = new node(incomingData);
			temp->prev = this->SP;
			this->SP = temp;
			this->capacity--;
		}
	}
	T pop() {
		if (this->isEmpty()) {
			std::cout << "StackUnderFlow!\n";
			return NULL;
		}
		else {
			node* temp = this->SP;
			this->SP = this->SP->prev;
			this->capacity++;
			return temp->data;
		}
	}
	void display() {
		this->stackPrinter(this->SP);
	}
	bool isFull() {
		if (this->capacity == 0) {
			return true;
		}
		else
			return false;
	}
	bool isEmpty() {
		if (this->capacity == this->size)
			return true;
		else
			return false;
	}
	void stackPrinter(node* temp) {
		if (temp == nullptr) {
			return;
		}
		std::cout << temp->data << "\n";
		temp = temp->prev;
		this->stackPrinter(temp);
	}
	T peek() {
		if (this->SP == nullptr)
			return NULL;
		return this->SP->data;
	}

};

int main()
{
	clock_t tStart = clock();
	Stack<MyObject> test(300000);
	for (int i = 0; i < 300001; i++)
	{
		test.push(MyObject(i + 1));
	}
	std::cout << test.peek() << "\n";
	for (int i = 0; i < 300001; i++)
	{
		test.pop();
	}
	test.display();
	std::cout << test.peek() << "\n";
	printf("Time taken: %.4fs\n", (double)(clock() - tStart) / CLOCKS_PER_SEC);
}
